packages/astro/src/core/dev/index.ts

```diff
@@ -275,6 +278,20 @@ export class AstroDevServer {
 		return viteServer;
 	}
 
+	private async stripDevRoot(req: http.IncomingMessage, res: http.ServerResponse, next: NextFunction) {
+		if (!this.viteServer) throw new Error(`AstroDevServer.start() not called`);
+		if (this.devRoot === '/') throw new Error(`AstroDevServer.stripDevRoot() called when there's no devRoot`);
+
+		let pathname = req.url || '/';
+		if (!pathname.startsWith(this.devRoot)) {
+			this.renderError(req, res, next);
+			return;
+		}
+		pathname = pathname.substr(this.devRoot.length) || '';
+		req.url = pathname.startsWith('/') ? pathname : '/' + pathname;
```

Comment:
Would this work to preserve slash?
```suggestion
		req.url = pathname.substr(this.devRoot.length - 1);
```